src/main/java/com/google/api/codegen/transformer/SchemaTypeTable.java

```diff
@@ -69,12 +69,12 @@ public String renderPrimitiveValue(Schema type, String value) {
   }
 
   @Override
-  public String getNicknameFor(Schema type) {
+  public String getNicknameFor(DiscoveryField type) {
```

Comment:
What is the motivation for switching from `Schema` to `DiscoveryField` everywhere?

Reply:
There has to be a central place that contains Schemas and their escaped, unique names. I wanted to have DiscoveryField handle getting unique names. Every transformer must go through DiscoveryField to get that base, unique name. This ensures that these Schema names are consistent throughout the pipeline. 

An alternative would to put all the name-escaping logic in Schema, but Schema seemed like too much of a pure model kind of class that was just data and no logic.